app/controllers/course/survey/responses_controller.rb

```diff
@@ -37,41 +37,33 @@ def edit
   end
 
   def update
-    if params[:response][:unsubmit]
-      handle_unsubmit
-    else
-      handle_update
-    end
-  end
-
-  private
-
-  def handle_create_error(error)
-    @response = @survey.responses.accessible_by(current_ability).
-                find_by(course_user_id: current_course_user.id)
-    if @response
-      render json: { responseId: @response.id }, status: :bad_request
+    @response.submit if params[:response][:submit]
+    if @response.update_attributes(response_update_params)
+      render_response_json
     else
-      render json: { error: error.message }, status: :bad_request
+      render json: { errors: @response.errors }, status: :bad_request
     end
   end
 
-  def handle_unsubmit
+  def unsubmit
     authorize!(:manage, @survey)
```

Comment:
yes, mind if it is updated in the next PR together with all the other abilities?